source/common/upstream/eds.cc

```diff
@@ -118,10 +119,34 @@ void EdsClusterImpl::onConfigUpdate(const Protobuf::RepeatedPtrField<ProtobufWkt
                                      cluster_load_assignment.cluster_name()));
   }
 
+  // Reset assignment timeout (if set) as we have received new assignment.
+  if (assignment_timeout_) {
+    assignment_timeout_->disableTimer();
+    assignment_timeout_.reset();
+  }
+  // Check if endpoint_stale_after is set.
+  const int64_t stale_after_ms =
+      PROTOBUF_GET_MS_OR_DEFAULT(cluster_load_assignment.policy(), endpoint_stale_after, 0);
+  if (stale_after_ms > 0) {
+    assignment_timeout_ = dispatcher_.createTimer([this]() -> void { onAssignmentTimeout(); });
```

Comment:
Rather than creating/deleting, I'd always just create the timer in the constructor and just do enable/disable here.

Reply:
I did this, but the test expectation started breaking (because of how the tests are written) and I was not able to resolve it.
Will take another stab at it. Wanted to see how strongly you felt about this.

Reply:
Done